automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/cluster/ClusterStatusTransitionsSpec.scala

```diff
@@ -48,6 +48,7 @@ class ClusterStatusTransitionsSpec extends GPAllocFixtureSpec with ParallelTestE
       // delete the cluster, but don't wait
       deleteCluster(billingProject, clusterName, monitor = false)
 
+      Thread.sleep(1000)
```

Comment:
Is this needed? I think the intent is that the cluster should be `Deleting` immediately after the `delete` call.

Reply:
since the queue processing is async to the API call now (after event published), DB might not have been updated yet right after the API call finishes

Reply:
Hm - so this is a change from our original behavior. Previously callers were guaranteed that the status is updated transactionally as part of calling a POST API. Now the status change may happen some time asynchronously _after_ the API returns.

Am wondering about possible effects here. For example when the UI pauses/deletes a runtime, it calls `POST` then `GET` immediately to get the new status. If there's now a race condition it may lead to delays in updating the status in the UI.

I understand why we did this: to recover better from pubsub publishing errors. But I'm wondering if there's a way to solve this while still making the status change transactional within the request. (Maybe we could error out the runtime instead if publishing fails or takes too long)

Reply:
This ties into the need for the `CLUSTER_PATCH` table, since a patch that requires a status transition doesn't happen transactionally with the request. 

I don't think we've implemented yet UI-side, but the goal is the UI will poll frequently when it GETs and sees the `isUpdating` flag.

We will definitely start getting complaints from the UI team and users if other endpoints start behaving inconsistently